search/search-client-query-transformer/src/main/java/io/camunda/search/clients/SearchClients.java

```diff
@@ -294,4 +298,42 @@ public SearchQueryResult<VariableEntity> searchVariables(final VariableQuery fil
             securityContext);
     return executor.search(filter, io.camunda.webapps.schema.entities.operate.VariableEntity.class);
   }
+
+  @Override
+  public Long countAssignees(final UsageMetricsQuery query) {
+    return distinctCountUsageMetricsFor("task_completed_by_assignee", query);
+  }
+
+  @Override
+  public Long countProcessInstances(final UsageMetricsQuery query) {
+    return distinctCountUsageMetricsFor("EVENT_PROCESS_INSTANCE_STARTED", query);
+  }
+
+  @Override
+  public Long countDecisionInstances(final UsageMetricsQuery query) {
+    return distinctCountUsageMetricsFor("EVENT_DECISION_INSTANCE_EVALUATED", query);
+  }
+
+  /*
+   * The distinct count is implemented here by using Java Stream API until aggregations are in place.
```

Comment:
❓ @tmetzke @ralfpuchert do we have aggregation support still planned for 8.7?

Reply:
They are planned for 8.8 according to [backlog](https://github.com/orgs/camunda/projects/126/views/1?filterQuery=-repo%3A%22camunda%2Fproduct-hub%22+aggre) : [issue](https://github.com/camunda/camunda/issues/21724)

Reply:
Well, I'd say it depends a bit. We might need "full" aggregation support for offering current V1 Operate internal endpoints in V2 (or at least partially). However, the aggregation we need here is basically an option in the search API to "count" and "count distinct". We could start with a really basic integration of this in the current search API so ES/OS and RDBMS can implement and map this operation accordingly.

When we later build a "proper" aggregation part of the search API, we can move this there if needed. But we should try to get "count" and "count distinct" in for 8.7 to make this a well-performing API in my view.

Disclaimer: this might be a lot more complex then I currently anticipate. However, I believe if we focus on those two options, we can get them in "pragmatically" without building a whole "concept" in the search API for now.

Reply:
Effectively using a filter aggregation on event name and obtaining the doc count from it should be suffice and we could potentially do that with a single query as well having one filter aggregation for each process, decision and task events querying both indices at the same time? Wdyt @ralfpuchert ?

I would be in favor to give that a try still before the 8.7 release, I could take a look next week in between the holidays if you don't have the capacity this week.

Reply:
I agree, that sounds like a feasible next iteration to try here. If you have time to look into it, @megglos, we'll definitely appreciate it 😅